### DISCUSSION TITLE
Read DTSU666 Energy Meter via Modbus RTU

### DISCUSSION THREAD

**Discussion by cresch06** [arendst/Tasmota#19237] [Support request]
I want to read the data of my DTSU666 Energy Meter via Modbus and send it via MQTT to home assistant.
What I have tried so far.
Wemos D1 mini with tasmota (v13.0) custom compiled firmeware (```#define USE_DDSU666```) and RS485 to TTL interface connected to smart meter. 
Data is displayed, unfortunately the registers of the DDSU666 do not match the registers of the DTSU666. Therefore the values are interpreted wrong.
The ModBus protocol address table for DTSU666 looks like this:
![PXL_20230731_142310882](https://github.com/arendst/Tasmota/assets/38973109/a1baa643-32dd-49b7-93ed-310dd239d05f)
![PXL_20230731_142317210~2](https://github.com/arendst/Tasmota/assets/38973109/3671c701-4aa7-4259-a5d0-75faa5b576ed)

The following two ways I  found so far for integrating the smart meter:
Based on the ```xnrg_11_ddsu666.ino``` make the necessary adjustments and write a ```xnrg_11_dtsu666.ino``` (the necessary changes should be limited). Programming skills are present on my part.

Or is it possible to address a new smart meter type via a json definition - have read the two documents development/tasmota/energy_modbus_configs/configurations.md and value_pair_descriptions.md.
The json string for the DTSU666 should start something like this:
```{"Name":"DTSU666", "Baud":"9600", "Config":"8N2", "Address":1, "Function":4, "Voltage":{"R":[0x2006H,0x2008H,0x200AH],"T":0,"F":-1}}```
What is unclear to me, where we pass this json string?

Which of the two variants is easier and better? For both I would also need some guidance.

Or is there another way, or is someone already working on the integration of a DTSU666?

**Comment by imod89** [comment]
Hi, 
have you found a solution in the meantime? 
I had the same idea of implementation and tried to get data of some DTSU666 via Tasmota Modbus to my MQTT server.

**Reply by pkkrusty** [reply]
Using the scripting smart meter driver is another option for pulling Modbus data. In some ways simpler than modifying the existing driver.
